src/utils.ts

```diff
@@ -68,3 +80,26 @@ export function useBase (base: string, handle: PHandle): PHandle {
     return handle(req, res)
   }
 }
+
+export function useQuery (req: IncomingMessage): URLSearchParams {
+  const url = new URL(req.url || '', 'http://localhost')
+  return url.searchParams
+}
+
+export function useBody<T> (request: IncomingMessage): Promise<T> {
+  return new Promise<T>((resolve, reject) => {
+    const body: any[] = []
+    request.on('error', (err) => {
+      reject(err)
+    }).on('data', (chunk) => {
+      body.push(chunk)
+    }).on('end', () => {
+      const bodyString = Buffer.concat(body).toString()
+      if (request.headers['content-type'] === 'application/json') {
```

Comment:
I would suggest split this part to `useJSON` / `getJSON` and destr for parsing

Reply:
You are suggesting to move headers check to another util? or just the parsing?
In the first scenario, what is the use-case of `useJSON`?

Reply:
Parsing body with destr regardless of headers (internally using useBody)